### PR TITLE
I tried to make interleave less tricky

### PR DESCRIPTION
**Pull request by UnixJunkie**

### CODE REVIEW

File: src/batList.mlv [hunk 1]
```diff
@@ -361,23 +361,19 @@ let takewhile = take_while
...
-  | ([h],  None,   None)       -> [h]
-  | ([h],  None,   Some x)     -> [h;x]
-  | ([h],  Some x, None)       -> [x;h]
-  | ([h],  Some x, Some y)     -> [x;h;y]
-  | (h::t, None  , None )      -> rev (aux [h] t)
-  | (h::t, Some x, None )      -> x::(rev (aux [h] t))
-  | (h::t, None,   Some y)     -> rev_append (aux [h] t) [y]
-  | (h::t, Some x, Some y)     -> x::rev_append (aux [h] t) [y]
+  let res = loop [] l in
+  may_prepend (rev (may_prepend res last)) first
```
Comment (gasche) [inline src/batList.mlv]: It is morally wrong to have the element be the *second* argument of `may_prepend` if it ends up to the *left* of the list in the result. You should write `let may_prepend maybe_elem list = ...`.

**Comment by UnixJunkie** [comment]
This one should be better.

**Comment by gasche** [comment]
Could you squash your two commits together to make for an easier post-merge history? (`git rebase -i HEAD~2`, then `git push -f` to your local branch, and github's going to update the PR automagically.)

**Comment by UnixJunkie** [comment]
I created another pull request then.
https://github.com/ocaml-batteries-team/batteries-included/pull/607

**Pull request closed without merging**